File: data/readable_format/fortran/file1308.f90
program test_appmth
    implicit none

    real :: runoff, remax, efflen, ealpha, m, effdrr, peakro

    ! Example Test Case
    runoff = 0.5
    remax = 1.0
    efflen = 100.0
    ealpha = 0.2
    m = 0.5
    effdrr = 10.0

    call appmth(runoff, remax, efflen, ealpha, m, effdrr, peakro)
    print *, "Fortran Test: Peak Runoff =", peakro

contains

    subroutine appmth(runoff,remax,efflen,ealpha,m,effdrr,peakro)

        real     runoff,efflen,ealpha,m,effdrr,remax,peakro
        real     tc

        real             te,vave,tstar,vstar,qpstar,a,b,c

        if (runoff.lt.0.00000001) then
            peakro = 0
            return
        endif
        vave  = runoff/effdrr
        te    = (efflen/(ealpha*vave**(m-1)))**(1./m)
        tstar = te/effdrr
        vstar = vave/remax

        if (tstar .ge. 1.) then
            qpstar = 1./tstar**m
        else
            if (vstar .lt. 1.) then
                a  = .6*(1.-vstar)
                b  = -1.
                c  = vstar
                tc = (-b-sqrt(b**2-4*a*c))/(2*a)

                if (tstar .gt. tc) then
                    qpstar = 1./tstar
                else
                    qpstar = 1./vstar - .6*(1.-vstar)/vstar * tstar
                end if
            else
                qpstar = 1.
            end if
        end if

        peakro = vave*qpstar

    end subroutine appmth

end program test_appmth

File: data/readable_format/cpp/file1308.cpp
#include <iostream>
#include <cassert>
#include <cmath> // For sqrt, pow, and fabs functions

void appmth(float runoff, float remax, float efflen, float ealpha, float m, float effdrr, float& peakro) {
    float tc;
    float te, vave, tstar, vstar, qpstar, a, b, c;

    if (runoff < 0.00000001f) {
        peakro = 0;
        return;
    }

    vave = runoff / effdrr;
    te = std::pow((efflen / (ealpha * std::pow(vave, m - 1.0f))), (1.0f / m));
    tstar = te / effdrr;
    vstar = vave / remax;

    if (tstar >= 1.0f) {
        qpstar = 1.0f / std::pow(tstar, m);
    } else {
        if (vstar < 1.0f) {
            a = 0.6f * (1.0f - vstar);
            b = -1.0f;
            c = vstar;
            tc = (-b - std::sqrt(b * b - 4.0f * a * c)) / (2.0f * a);

            if (tstar > tc) {
                qpstar = 1.0f / tstar;
            } else {
                qpstar = 1.0f / vstar - 0.6f * (1.0f - vstar) / vstar * tstar;
            }
        } else {
            qpstar = 1.0f;
        }
    }

    peakro = vave * qpstar;
}

int main() {
    float runoff = 0.5f;
    float remax = 1.0f;
    float efflen = 100.0f;
    float ealpha = 0.2f;
    float m = 0.5f;
    float effdrr = 10.0f;
    float peakro;

    appmth(runoff, remax, efflen, ealpha, m, effdrr, peakro);
    std::cout << "C++ Test: Peak Runoff = " << peakro << std::endl;

    // Example assertion based on expected value from Fortran output for this test case
    // Adjust the expected value based on your Fortran program's output
    assert(std::fabs(peakro - 0.00141421368) < 0.0001);

    std::cout << "All tests passed." << std::endl;
    return 0;
}